utils/utils.go

```diff
@@ -491,6 +499,82 @@ func GetAccountBalances(
 	return accountBalances, nil
 }
 
+// -------------------------------------------------------------------------------
+// ----------------- Helper struct for fetching account coins --------------------
+// -------------------------------------------------------------------------------
+
+// AccountCoinsRequest defines the required information to get an account's coins.
+type AccountCoinsRequest struct {
+	Account        *types.AccountIdentifier
+	Network        *types.NetworkIdentifier
+	Currencies     []*types.Currency
+	IncludeMempool bool
+}
+
+// AccountCoins defines an account's coins info in the tip.
+type AccountCoinsResponse struct {
+	Coins []*types.Coin
+}
+
+// GetAccountCoins calls /account/coins endpoint and returns an array of coins in the tip.
+func GetAccountCoins(
+	ctx context.Context,
+	fetcher FetcherHelper,
+	acctCoinsReqs []*AccountCoinsRequest,
+) ([]*AccountCoinsResponse, error) {
+	var acctCoins []*AccountCoinsResponse
+	for _, req := range acctCoinsReqs {
+		coins, err := getAccountCoins(
+			ctx,
+			fetcher,
+			req.Account,
+			req.Currencies,
+			req.IncludeMempool,
+			req.Network,
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		resp := &AccountCoinsResponse{
+			Coins: coins,
+		}
+
+		acctCoins = append(acctCoins, resp)
+	}
+
+	return acctCoins, nil
+}
+
+// getAccountCoins returns Rosetta type of response from /account/coins endpoint.
+func getAccountCoins(
```

Comment:
Why we need this extra function?

Reply:
I think you are right, I'll remove this function.

Reply:
Done